#### Домашняя работа №2


##### Задание №1

1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. 
Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [1]:
from math import factorial as facto
from math import exp

In [2]:
k = 85
p = 0.8
n = 100

c = facto(n) / (facto(k) * facto(n - k))
P = c * (p**k) * (1-p)**(n - k)
P

0.048061793700746355

Ответ: вероятность того, что стрелок попадёт в цель ровно 85 раз Р = 0.04806

##### Задание №2

2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

In [3]:
n = 5000
p = 0.0004
m0 = 0
m2 = 2

lambdaa = n * p

Puasson_0 = (lambdaa**m0 / facto(m0)) * exp(-lambdaa)
Puasson_2 = (lambdaa**m2 / facto(m2)) * exp(-lambdaa)
Puasson_0, Puasson_2

(0.1353352832366127, 0.2706705664732254)

Ответ: 1) вероятность того, что ни одна из них не перегорит в первый день Р = 0.13533
       2) вероятность того, что перегорят ровно две Р = 0.27067

##### Задание №3

3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [4]:
k = 70
p = 0.5
n = 144

c = facto(n) / (facto(k) * facto(n - k))
P = c*(p**k) * (1 - p)**(n - k)
P

0.06281178035144776

Ответ: вероятность того, что выпадет орёл ровно 70 раз Р = 0.06281

##### Задание №4

4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. 

а) Какова вероятность того, что все мячи белые? 

б) Какова вероятность того, что ровно два мяча белые? 

в) Какова вероятность того, что хотя бы один мяч белый?

In [5]:
n1 = 10                    # количество мячей в 1-ом ящике
m1 = 7                     # количество белых мячей в 1-ом ящике
n2 = 11                    # количество мячей в 2-ом ящике
m2 = 9                     # количество белых мячей в 2-ом ящике

In [6]:
# а) Какова вероятность того, что все мячи белые?

P1 = (m1 / n1) * ((m1 - 1) / (n1 - 1))     # 1-ой ящик
P2 = (m2 / n2) * ((m2 - 1) / (n2 - 1))     # 2-ой ящик
P = P1 * P2
P

0.3054545454545455

In [7]:
# б) Какова вероятность того, что ровно два мяча белые?

# Возможны три варианта, когда два мяча белые:
# - 2 белых мяча из первого ящика (P21)
 
P1_1 = (m1 / n1) * ((m1 - 1) / (n1 - 1))      # вероятность того, что из 1-го ящика вытащили подряд 2 белых мяча 
P1_2 = (n2 - m2) / n2                         # вероятность того, что из 2-го ящика вытащили подряд 2 черных мяча
P21 = P1_1 * P1_2
P21

0.08484848484848484

In [8]:
# - ровно по 1 из каждого ящика (P11)

k = 1
n = 2
p1 = 7/10
p2 = 9/11

C = facto(n) / (facto(k) * facto(n - k))

P2_1 = C * (p1**k) * (1 - p1)**(n - k)
P2_2 = C * (p2**k) * (1 - p2)**(n - k)
P11 = P2_1 * P2_2
P11

0.12495867768595041

In [9]:
# - 2 белых из второго ящика (P22)

P3_1 = (m2 / n2) * ((m2 - 1) / (n2 - 1))    # вероятность того, что из 2-го ящика вытащили подряд 2 белых мяча
P3_2 = (n1 - m1) / n1                       # вероятность того, что из 1-го ящика вытащили подряд 2 черных мяча
P22 = P3_1 * P3_2
P22

0.19636363636363638

In [10]:
P = P21 * P11 * P22
P

0.0020819561505361654

In [11]:
# в) Какова вероятность того, что хотя бы один мяч белый?

P = (m1 + m1) / (n1 + n2)
P

0.6666666666666666

Ответ: а) вероятность того, что все мячи белые Р = 0.30545
       б) вероятность того, что ровно два мяча белые Р = 0.00208
       в) вероятность того, что хотя бы один мяч белый Р = 0.66666